Automattic/jetpack
# Feature/infinite scroll older twenty themes

This PR extends the AMP Infinite Scroll support #16048 for older `twenty` themes.

#### Changes proposed in this Pull Request:

* This PR extends mobile AMP Infinite Scroll support for older `twenty` themes.

#### Jetpack product discussion


* This extends support for Infinite Scroll on AMP requests;

#### Does this pull request change what data or activity we track or use?


* No

#### Testing instructions:






1. Ensure the AMP plugin is installed and enabled;
1. On Jetpack Settings > Writing > Theme Enhancements > Infinite Scroll, set the radio to either `Load more posts on a page with a button` or `Load more posts as the reader scrolls down`. Due to[ AMP implementation](https://amp.dev/documentation/components/amp-next-page/#recommendation-box-(more-suggested-links)) of the functionality, the next page will always be autoloaded and still show the Older Posts button, if the user scrolls faster then the next page loads;
1. Enable an older `twenty` theme;
1. Make sure you have enough posts to enable pagination;
1. Visit the home page or an archive page on mobile mode;
1. Scroll to the footer. The next page should be loaded by then, or you can click the older posts button;
#### Proposed changelog entry for your changes:

* Adds mobile support for Infinite Scroll on older `twenty` themes for AMP requests:
  * `twentyten`;
  * `twentyeleven`;
  * `twentytwelve`;
  * `twentythirteen`;
  * `twentyfourteen`;
  * `twentyfifteen`;
  * `tweentysixteen`;
  * `twentyseventeen`;
* Due to technical limitations, there are significant markup changes that sacrifice the desktop experience with these themes.
